src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -90,26 +83,27 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
         }
 
         // dispatch to the correct handler
-        if (objResult.getsObjectType() == Allocation__c.sObjectType && !hasRunForAllocationsBefore && 
+        if (objResult.getsObjectType() == Allocation__c.sObjectType && !TDTM_ProcessControl.checkRecursionState(TDTM_ProcessControl.flag.ALLOC) &&
             (triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate)) {
```

Comment:
Given the trigger and dml flow I'm not seeing why it has to track the before/after/opportunity operations separately as 3 different flags. Not that I can't reimplement that logic, but it appears to be unnecessary. I'm very curious what @njjc thinks about these 3 flags.